javascript/diff_match_patch_uncompressed.js

```diff
@@ -2397,16 +2386,22 @@ diff_match_patch.patch_obj.prototype.toString = function() {
 
 // CLOSURE:begin_strip
 // Lines below here will not be included in the Closure-compatible library.
-
-// Export these global variables so that they survive Google's JS compiler.
-// In a browser, 'this' will be 'window'.
-// Users of node.js should 'require' the uncompressed version since Google's
-// JS compiler may break the following exports for non-browser environments.
-/** @suppress {globalThis} */
-this['diff_match_patch'] = diff_match_patch;
-/** @suppress {globalThis} */
-this['DIFF_DELETE'] = DIFF_DELETE;
-/** @suppress {globalThis} */
-this['DIFF_INSERT'] = DIFF_INSERT;
-/** @suppress {globalThis} */
-this['DIFF_EQUAL'] = DIFF_EQUAL;
+if (typeof module === 'object') {
+  // The following export code was added by @ForbesLindesay
+  module.exports = diff_match_patch;
+  module.exports['diff_match_patch'] = diff_match_patch;
+  module.exports['DIFF_DELETE'] = DIFF_DELETE;
+  module.exports['DIFF_INSERT'] = DIFF_INSERT;
+  module.exports['DIFF_EQUAL'] = DIFF_EQUAL;
+} else {
+  // Export these global variables so that they survive Google's JS compiler.
+  // In a browser, 'this' will be 'window'.
+  /** @suppress {globalThis} */
+  this['diff_match_patch'] = diff_match_patch;
+  /** @suppress {globalThis} */
+  this['DIFF_DELETE'] = DIFF_DELETE;
+  /** @suppress {globalThis} */
+  this['DIFF_INSERT'] = DIFF_INSERT;
+  /** @suppress {globalThis} */
+  this['DIFF_EQUAL'] = DIFF_EQUAL;
```

Comment:
I could see this section being a potentially good first patch all by itself for this change. as in, for step one, could we leave everything else alone and make the module work in the existing setup _and_ make it possible for people to import it directly as an esmodule?

Reply:
I can pull this out to a new PR if you like. It will help with CommonJS deployments, but the diff API will still have issues in an ES6 environment as I mentioned in my other comment.

Reply:
could we leave it as ESModule syntax in the code and have our build tool generate a CommonJS module?

if we _break_ with the vendored file of the past I don't see why we'd need to make Closure compiler happy

Reply:
From what I've seen, build tools seem to do just something like the `index.mjs` file in this PR. Projects that support both do it as I've done in this PR, although I certainly won't complain if you want to make it ESM-only.